core/statistics.c

```diff
@@ -119,17 +123,24 @@ void calculate_stats_summary(struct stats_summary *out, bool selected_only)
 
 	size = sizeof(stats_t) * (dive_table.nr + 1);
 	tsize = sizeof(stats_t) * (NUM_DIVEMODE + 1);
+	dsize = sizeof(stats_t) * ((STATS_MAX_DEPTH/STATS_DEPTH_BUCKET) + 1);
+	tmsize = sizeof(stats_t) * ((STATS_MAX_TEMP/STATS_TEMP_BUCKET) + 1);
 	free_stats_summary(out);
 	out->stats_yearly = malloc(size);
 	out->stats_monthly = malloc(size);
 	out->stats_by_trip = malloc(size);
 	out->stats_by_type = malloc(tsize);
-	if (!out->stats_yearly || !out->stats_monthly || !out->stats_by_trip || !out->stats_by_type)
+	out->stats_by_depth = malloc(dsize);
+	out->stats_by_temp = malloc(tmsize);
+	if (!out->stats_yearly || !out->stats_monthly || !out->stats_by_trip ||
+		  !out->stats_by_type || !out->stats_by_depth || !out->stats_by_temp)
```

Comment:
Also in original code: We're in a OOM situation here so this will never happen. But shouldn't this do something like
```
free(*out);
*out = NULL;
```
?

Reply:
Answering to my self here: No, not needed. The caller will do that using the C++ wrapper.